utils: Add the `omnia-eeprom` utility

elkablo opened the pull request:
Add a new utility, `omnia-eeprom`, which can be used to print / set EEPROM fields on Turris Omnia.

One example when this utility might be useful is if the board experiences random crashes due to newer versions of the DDR training algorithm in newer U-Boot. The user can change the DDR speed from 1600K to 1333H to solve these issues, with

  ```
  omnia-eeprom set ddr_speed 1333H
  ```

robimarko commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,55 @@
+endef
+
+define Build/Compile
+	$(MAKE) -C $(PKG_BUILD_DIR) \
```
I would suggest just extending `Build/Compile/Default` instead of a custom `Build/Compile` recipe

drujd commented:
This needs the eeprom-at24 kernel module. It should be added to dependencies.

robimarko commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,54 @@
+
+PKG_NAME:=omnia-eeprom
+PKG_VERSION:=v0.1
+PKG_RELEASE:=$(AUTORELEASE)
```
`AUTORELEASE` has been deprecated and should not be used.

BKPepe replied:
Yeah, easy to fix to apply this suggestion:
```suggestion
PKG_RELEASE:=1
```

BKPepe commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,54 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=omnia-eeprom
+PKG_VERSION:=v0.1
```
I think that your versioning is not compatible with APK.

It should be without the v before the version:
```suggestion
PKG_VERSION:=0.1
```

Unfortunately, you need to tweak other variables.

BKPepe commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,54 @@
+PKG_RELEASE:=$(AUTORELEASE)
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_SOURCE_URL:=https://gitlab.nic.cz/turris/$(PKG_NAME)/-/archive/$(PKG_VERSION)/
```
```suggestion
PKG_SOURCE_URL:=https://gitlab.nic.cz/turris/omnia-eeprom/-/archive/$(PKG_VERSION)/
```

This makes it more readable and someone can copy it to the address bar.

BKPepe commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,54 @@
+define Package/omnia-eeprom
+  SECTION:=utils
+  CATEGORY:=Utilities
+  URL:=https://gitlab.nic.cz/turris/$(PKG_NAME)
```
```suggestion
  URL:=https://gitlab.nic.cz/turris/omnia-eeprom
```

This makes it easy to copy and paste it to the address bar.

BKPepe commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,54 @@
+the DDR training algorithm).
+endef
+
+MAKE_VARS += \
+	OMNIA_EEPROM_VERSION="$(PKG_VERSION)"
```
```suggestion
MAKE_VARS += OMNIA_EEPROM_VERSION="$(PKG_VERSION)"
```

BKPepe commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,54 @@
+PKG_HASH:=6f949d0b8080adca8bae088774ce615b563ba6ec2807cce97ee6769b4eee7bbf
+
+PKG_MAINTAINER:=Marek Behun <kabel@kernel.org>
+PKG_LICENSE:=GPL-2.0
```
Since this is deprecated SPDX License identifier, you need to know if it is GPL 2.0 only or later according to https://spdx.org/licenses/

BKPepe commented on package/utils/omnia-eeprom/Makefile:
```diff
@@ -0,0 +1,54 @@
+define Package/omnia-eeprom/description
+This package contains the omnia-eeprom utility, which allows you to display
+and update EEPROM fields on the Turris Omnia router.
+The EEPROM is normally not meant to be updated by users, but there are some
+exceptions where it might be useful.
```
I am thinking loud. Do we really need to have this package preinstalled for all Turris Omnia routers? I know they are huge in every spec than usual OpenWrt router, but for now, the only benefit of this package is that it helps your customers who have issues with recent U-boot loader on their older router revisions. But if the user is updating U-boot, he is skilled enough and knows what he does, thus shouldnt this be in the packages feed?

elkablo replied:
The issue this package is meant to solve is *random crash due to DDR errors*. If a router ends up in this state, we want to have the package available. Having to download the package and install it takes some CPU computations and network controller accesses to DDR, which just makes it more probable that a DDR error will occur and the router will crash before the user is able to execute `omnia-eeprom`.

Combined with the fact that this package compiles to a 12.5 KiB binary, I think it can safely be installed for all Omnias.

elkablo committed:
utils: Add the omnia-eeprom utility

Add a new utility, `omnia-eeprom`, which can be used to print / set
EEPROM fields on Turris Omnia.

One example when this utility might be useful is if the board
experiences random crashes due to newer versions of the DDR training
algorithm in newer U-Boot. The user can change the DDR speed from 1600K
to 1333H to solve these issues, with

  ```
  omnia-eeprom set ddr_speed 1333H
  ```

Signed-off-by: Marek Behún <kabel@kernel.org>
Link: https://github.com/openwrt/openwrt/pull/16264
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Any reason this is still a draft?

elkablo commented:
@robimarko I marked this ready for review just now. Waited for last compile test.

robimarko commented:
Thanks! Rebased on top of main and merged!
